### PR TITLE
Graduate server side validation to beta

### PR DESCRIPTION
**Pull request by kevindelgado**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Graduate server side field validation to beta and enable it by default.

All kubectl requests that set `--validate=true` by default will do so via strict server-side rather than client-side (for all requests targeting clusters with this feature enabled).

The apiserver will default to warning on unknown/invalid fields unless strict or ignore validation is requested.

#### Which issue(s) this PR fixes:

Refers to: https://github.com/kubernetes/enhancements/issues/2885

#### Does this PR introduce a user-facing change?

```release-note
Graduated ServerSideFieldValidation to `beta`. Schema validation is performed server-side and requests will receive warnings for any invalid/unknown fields by default.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
KEP: https://github.com/kubernetes/enhancements/issues/2885
```

### CODE REVIEW

**Comment by kevindelgado** [comment]
This PR is based on top of https://github.com/kubernetes/kubernetes/pull/109494, only the latest 2 commits are new

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1045,8 +1045,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	genericfeatures.ServerSideApply: {Default: true, PreRelease: featuregate.GA},
 
-	genericfeatures.ServerSideFieldValidation: {Default: false, PreRelease: featuregate.Alpha},
-
+	genericfeatures.ServerSideFieldValidation: {Default: true, PreRelease: featuregate.Beta},
```
Comment (liggitt) [inline pkg/features/kube_features.go:1048]: nit: keep the newline after this (keeping them separate keeps indentation independent)

File: test/e2e/kubectl/kubectl.go [hunk 2]
```diff
@@ -1185,12 +1194,13 @@ metadata:
...
 		ginkgo.It("should detect unknown metadata fields of a typed object", func() {
 			ginkgo.By("calling kubectl create deployment")
-			invalidMetaDeployment := `
+			invalidMetaDeployment := fmt.Sprintf(`
 	{
 		"apiVersion": "apps/v1",
 		"kind": "Deployment",
 		"metadata": {
 			"name": "my-dep",
+			"namespace": "%s",
```
Comment (liggitt) [inline test/e2e/kubectl/kubectl.go:1203]: was this change required? I don't remember this bit

**Review by kevindelgado — COMMENTED** [review]
PTAL @liggitt

File: test/e2e/kubectl/kubectl.go (hunk 2 above)
Comment (liggitt) [reply test/e2e/kubectl/kubectl.go:1203]: oh, huh... should it have been added in https://github.com/kubernetes/kubernetes/pull/110509 then? I just wanted to make sure it wasn't newly required because of the server-side field validation feature defaulting on

if this test passes without this change, let's hoist this namespace addition to a separate PR to avoid entangling an unrelated fix in the feature enablement PR

Comment (kevindelgado) [reply test/e2e/kubectl/kubectl.go]: Yea, I've removed it.

I'm trying to remember if it was actually needed or not. I went through the PR CI history, and it doesn't seem like the test was actually breaking because of this. 

Also, looking at #110509, the error is from hardcoding the namespace not omitting it. If this PR passes without it, I don't think we need it.
 
I think maybe I just got nervous seeing a kubectl call in an e2e test I was working on without the namespace set and just added it to be safe, but I don't think it's actually necessary

**Comment by liggitt** [comment]
last two commits (feature enablement and test addition) this PR adds on top of #109494 lgtm

/lgtm
/approve

**Comment by liggitt** [comment]
/retest

kubectl wait unit test flake is an unrelated known issue
